internal/cache/util/util.go

```diff
@@ -99,3 +101,25 @@ func IsCacheHandleInvalid(readErr error) bool {
 		strings.Contains(readErr.Error(), ErrInSeekingFileHandleMsg) ||
 		strings.Contains(readErr.Error(), ErrInReadingFileHandleMsg)
 }
+
+func CheckCacheDirectoryPermissions(dirPath string) (bool, error) {
+	_, statErr := os.Stat(dirPath)
+
+	if statErr != nil {
+		err := os.MkdirAll(dirPath, FileDirPerm)
+		if err != nil {
+			return false, fmt.Errorf("error in creating directory structure %s: %v", dirPath, err)
+		}
+		return true, nil
+	}
+
+	f, err := fsutil.AnonymousFile(dirPath)
```

Comment:
fsutil.AnonymousFile has the following warning:
```
// Warning: this is not production-quality code, and should only be used for
// testing purposes. In particular, there is a race between creating and
// unlinking by name.
```

Why didn't we check for directory permissions instead?

Reply:
Didn't know about the warning associated with it, however the warning is not applicable to our use case here and fsutil.AnonymousFile is also used in write flow of GCSFuse https://github.com/GoogleCloudPlatform/gcsfuse/blob/2d635761ba0baa584d56be4332835db213b65404/internal/fs/inode/file.go#L258, so we should be fine to use it here.

@ankitaluthra1 can answer to the question put in the comment.

Reply:
This is being used in checking temp dir access which is live in prod, its okay to ignore this warning imo

Reply:
permission check is tricky, as we wont be sure if any umask is applied or with what type of user gcsfuse has been  mounted (like same user could be owner or other in different scenario), creating a file is sureshot way to know if we can create a file with current user.